Fix for things upstream’s configure.ac does wrong:
• mis-placed closing parenthesis for AC_HELP_STRING prevents
  the AC_ARG_WITH(ssh, …) from actually doing anything
• check for existence, not declaration, of __progname

Index: autossh/configure.ac
===================================================================
--- autossh.orig/configure.ac	2011-06-05 00:10:01.000000000 +0200
+++ autossh/configure.ac	2011-06-05 00:26:02.000000000 +0200
@@ -16,9 +16,8 @@
 fi
 
 AC_ARG_WITH(ssh,
-	AC_HELP_STRING([--with-ssh=ARG], [specify path to ssh executable],
-			[ac_cv_path_ssh=$withval], []
-	)
+	AC_HELP_STRING([--with-ssh=ARG], [specify path to ssh executable]),
+	[ac_cv_path_ssh=$withval], []
 )
 if test -z "$ac_cv_path_ssh"; then
 	AC_PATH_PROG([ssh], [ssh])
@@ -118,9 +117,29 @@
 AC_CHECK_LIB(socket, connect)
 
 # Other checks
-AC_CHECK_DECL(__progname,
-	AC_DEFINE(HAVE___PROGNAME, 1,
-	[Define to 1 if __progname is available.]),
-)
+
+# Copyright (c) 1999-2004 Damien Miller
+#
+# Permission to use, copy, modify, and distribute this software for any
+# purpose with or without fee is hereby granted, provided that the above
+# copyright notice and this permission notice appear in all copies.
+#
+# THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+# WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+# MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+# ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+# WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+# ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+# OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+AC_CACHE_CHECK([if libc defines __progname], ac_cv_libc_defines___progname, [
+	AC_TRY_LINK([],
+		[ extern char *__progname; printf("%s", __progname); ],
+		[ ac_cv_libc_defines___progname="yes" ],
+		[ ac_cv_libc_defines___progname="no" ]
+	)
+])
+if test "x$ac_cv_libc_defines___progname" = "xyes" ; then
+	AC_DEFINE(HAVE___PROGNAME, 1, [Define if libc defines __progname])
+fi
 
 AC_OUTPUT
